### PR TITLE
Fixes to the syntax highlighting mode code

### PR DESCRIPTION
**Pull request by jdfreder**

### CODE REVIEW

File: src/notebook/actions/index.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ export function setExecutionState(executionState) {
 export function setLanguageInfo(langInfo) {
   return {
     type: constants.SET_LANGUAGE_INFO,
-    langInfo,
+    langInfo: Immutable.fromJS(langInfo),
```
Comment (jdfreder) [inline src/notebook/actions/index.js]: @rgbkrk there was mutable state in our app state 😲

Comment (rgbkrk) [inline src/notebook/actions/index.js]: Isn't this an action creator though? Wouldn't the reducer be handling the immutable conversion?

Comment (rgbkrk) [inline src/notebook/actions/index.js]: Now I realize that some of our other action creators use commutable directly as well. Here I thought we wanted the actions to be able to be sent over the wire.

File: src/notebook/components/notebook.js [hunk 2]
```diff
@@ -147,17 +136,44 @@ class Notebook extends React.Component {
...
+    let language =
+      this.props.notebook.getIn(['metadata', 'language_info', 'codemirror_mode', 'name'],
+      this.props.notebook.getIn(['metadata', 'language_info', 'codemirror_mode'],
+      this.props.notebook.getIn(['metadata', 'language_info', 'name'],
+      'text')));
+
+    // TODO: Load the ipython codemirror mode from somewhere
+    if (language === 'ipython') {
+      language = 'python';
+    }
```
Comment (rgbkrk) [inline src/notebook/components/notebook.js:151]: 👍 to all this.

File: src/notebook/actions/index.js (hunk 1 above)
Comment (jdfreder) [inline src/notebook/actions/index.js]: Yeah :(

**Comment by rgbkrk** [comment]
I'm going to give you a pass on the codecov and fully expect us to run tests against the high level notebook component soon. :smile:

**Pull request merged**